src/main/java/org/apache/tomcat/jakartaee/ManifestConverter.java

```diff
@@ -48,20 +48,16 @@ public boolean accepts(String filename) {
     }
 
     @Override
-    public void convert(String path, InputStream src, OutputStream dest, EESpecProfile profile) throws IOException {
+    public boolean convert(String path, InputStream src, OutputStream dest, EESpecProfile profile) throws IOException {
         Manifest srcManifest = new Manifest(src);
         Manifest destManifest = new Manifest(srcManifest);
 
-        boolean result = false;
-
-        result = result | removeSignatures(destManifest);
+        boolean result = removeSignatures(destManifest);
         result = result | updateValues(destManifest, profile);
 
-        if (result) {
-            destManifest.write(dest);
-        } else {
-            srcManifest.write(dest);
```

Comment:
In testing, I found that writing the source manifest doesn't guarantee the order of attributes, so the manifest differs even with no value changes (comparing using `pkgdiff`). This and the change below allows the implementation version to change only when other conversions were also made.

Reply:
This change raises the problem that if a version number exists, it must not be the original version number, even if the rest of the content has not been converted.

Reply:
Here it is still necessary to revert, as the conversion is fine for the specified jar, but for directory migration via the migrator, the manifest of some jars here will be affected(some jars are converted, some are not, but information for the unconverted jars is changed), even if the jar is not converted.

Reply:
> some jars are converted, some are not, but information for the unconverted jars is changed

`Attributes` is a `HashMap` so in the majority of cases, reading and writing the manifest in this way is causing the key order to change, which is why I switched to always writing `destManifest`. If the intention was to write the manifest unchanged, we'd write original bytes instead, and use another way to determine if only implementation version was changed for the conversion return value.

Reply:
Pushed a new commit which should address these concerns: https://github.com/apache/tomcat-jakartaee-migration/pull/36/commits/136c537e1e9d0274261d8749fe3a6747f6a9dc3c.